### DISCUSSION TITLE
Six-segment cron expressions

### DISCUSSION THREAD

**Discussion by GeoffArmstrong** [jamesmh/coravel#367] [General]
I have a number of use cases where six segment cron expressions are a big step up from five segment cron expressions (jobs I currently run using Azure Webjobs). The six segments being: second, minute, hour, day, month, and weekday.  I have jobs that I want to execute at particular times during each minute (e.g. 30 seconds after the start of the minute).  Is this something that coravel supports, or is it just five segment cron expressions?

**Comment by jamesmh** [comment]
Hey Geoff, 

Coravel doesn't support setting a raw 6 segment cron expression, but you can do seconds-based scheduling like:

```csharp
scheduler
    .Schedule<MyInvocable>()
    .EveryThirtySeconds();
```

There are further restrictions using the fluent API that you can do too like:

```csharp
scheduler
    .Schedule<MyInvocable>()
    .EveryThirtySeconds()
    .Weekend();
```

So this would only run on the 30 second mark of the minute during weekend hours (based on UTC times), for example.

For your use case, I think it would depend on how complex your cron expressions are and whether those translate into what the fluent API offers.

**Reply by GeoffArmstrong** [reply]
Thanks for your reply!  I really like what I see in Coravel and if I end up bringing it into my projects I'd probably modify the code to add support for 6-segment cron expressions and offer a pull request.

I can probably do the stuff I need with the Fluent API - FYI here are the cron expressions that I use:

0 0 2 8-14 * 2 = run 2:00 AM on the 2nd Tuesday of the month.
30 15 */8 * * * = run 30 seconds past the minute, at 15 minutes past the hour, every 8 hours.
30 * * * * * = run 30 seconds past the minute every minute

**Reply by jamesmh** [reply]
Cool - looks like the 2nd one would be the issue then eh? The 1st and 3rd should have a fluent API equivalent 👍
